ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -99,7 +106,9 @@
   private Offset prevSafeEndPoint = null;
   // reset key refers to the first StoreKey that is added to the index segment
   private Pair<StoreKey, PersistentIndex.IndexEntryType> resetKey = null;
-  protected ConcurrentSkipListMap<StoreKey, IndexValue> index = null;
+  private ConcurrentSkipListMap<StoreKey, NavigableSet<IndexValue>> index = null;
```

Comment:
Because I now believe the associated set should be concurrent, this should be
`ConcurrentSkipListMap<StoreKey, ConcurrentSkipListSet<IndexValue>>`

Reply:
Done